### PR TITLE
Upstreaming OpenBSD local changes from the port.

### PR DESCRIPTION
**Pull request by devnexen**
- By default disabling JIT to please the W^X policy.
- Little missing headers/little code changes.
- In addition, clock_gettime test is somewhat flaky due to
this cmake helper thus checking code-wise instead.

### CODE REVIEW

File: libraries/zmusic/mididevices/music_fluidsynth_mididevice.cpp [hunk 1]
```diff
@@ -140,8 +140,12 @@ class FluidSynthMIDIDevice : public SoftSynthMIDIDevice
 #define FLUIDSYNTHLIB1	"libfluidsynth.1.dylib"
 #define FLUIDSYNTHLIB2	"libfluidsynth.2.dylib"
 #else // !__APPLE__
+#ifndef FLUIDSYNTHLIB1
 #define FLUIDSYNTHLIB1	"libfluidsynth.so.1"
+#endif
+#ifndef FLUIDSYNTHLIB2
 #define FLUIDSYNTHLIB2	"libfluidsynth.so.2"
+#endif
```
Comment (alexey-lysiuk) [inline libraries/zmusic/mididevices/music_fluidsynth_mididevice.cpp]: What's the purpose of this change? If it's about providing custom library names, then why is it for FluidSynth only?

File: CMakeLists.txt [hunk 2]
```diff
@@ -183,7 +183,10 @@ include( TargetArch )
 target_architecture(ZDOOM_TARGET_ARCH)
 
 if( ${ZDOOM_TARGET_ARCH} MATCHES "x86_64" )
-	set( HAVE_VM_JIT ON )
+	# To make the game respecting OpenBSD's W^X policy by default
+	if(NOT CMAKE_PLATFORM_NAME MATCHES "OpenBSD")
+		set( HAVE_VM_JIT ON )
+	endif()
```
Comment (alexey-lysiuk) [inline CMakeLists.txt]: In my opinion, AsmJit needs to be adapted to such memory protection policy instead of disabling JIT completely.
Most likely, [this fragment](https://github.com/coelckers/gzdoom/blob/e0fb9a45e21daba6afa40ef192aa7d9e3923ef20/libraries/asmjit/asmjit/base/osutils.cpp#L134-L138) should be extended to never have write and execute permissions at the same time.

Comment (devnexen) [reply CMakeLists.txt]: I read somewhere in here it is already a heavily modified version of asmjit, might be better some upstreaming job done first ?

File: libraries/zmusic/mididevices/music_fluidsynth_mididevice.cpp (hunk 1 above)
Comment (devnexen) [reply libraries/zmusic/mididevices/music_fluidsynth_mididevice.cpp]: Because it is only needed for these.

File: CMakeLists.txt (hunk 2 above)
Comment (dpjudas) [reply CMakeLists.txt]: Upstreaming our asmjit changes will most likely not happen. The upstream author doesn't seem particular interested in the features in question (or he's too busy with other things to deal with them) and I don't think I can create a proper PR that would be accepted upstream.

**Comment by alexey-lysiuk** [comment]
Closing this as it needs to be split into two PRs, to ZMusic and GZDoom repositories.
